src/actions/guides/frontend/html_forms.cr

```diff
@@ -0,0 +1,533 @@
+    ```
+
+    ```crystal
+    checkbox(op.with_cheese, "no", "yes", class: "custom-check")
+    ```
+
+    ### Radio
+
+    ```html
+    <input type="radio"
+           name="param_key:question_five"
+           value="Yes" />
+    <input type="radio"
+           name="param_key:question_five"
+           value="No" />
+    ```
+
+    ```crystal
+    input(op.question_five, type: "radio", name: "\#{op.param_key}:\#{op.question_five.name}", value: "Yes")
+    input(op.question_five, type: "radio", name: "\#{op.param_key}:\#{op.question_five.name}", value: "No")
+    ```
+
+    > ISSUE REF: https://github.com/luckyframework/lucky/issues/1023
+
+    ## Buttons
+
+    ### button tag
+
+    ```html
+    <button role="submit" class="btn">Go!</button>
+    ```
+
+    ```crystal
+    button("Go!", role: "submit", class: "btn")
+    ```
+
+    ### submit input
+
+    ```html
+    <input type="submit" value="Go!" class="btn" />
+    ```
+
+    ```crystal
+    submit("Go!", class: "btn")
+    ```
+
+    ## Labels
+
+    ```html
+    <label for="param_key_first_name" class="custom-label">First Name</label>
+    ```
+
+    ```crystal
+    label_for(op.first_name)
+
+    # or with custom text
+
+    label_for(op.first_name, "Enter your First name:")
+
+    # or using a block
+
+    label_for(op.first_name) do
+      text("First Name: ")
+    end
+    ```
+
+    ## Saving Data
+
+    HTML Forms in Lucky are based around the concept of [Operations](#{Guides::Database::ValidatingSaving.path}). We use these for securing param values from form inputs, and doing validations.
+
+    For info on interacting with databases, see the [saving
+    data with operations](#{Guides::Database::ValidatingSaving.path(anchor: Guides::Database::ValidatingSaving::ANCHOR_USING_WITH_HTML_FORMS)}) guide.
+
+    ## Displaying Errors
+
+    If your permitted column / attribute fail any sort of validation, it's common to
+    display an error about why it failed. For this, Lucky generates a component you can use
+    in `src/components/shared/field_errors.cr`.
+
+    ```html
+    <div class="error">Email must be valid</div>
+    ```
+
+    ```crystal
+    mount Shared::FieldErrors.new(op.email)
+    ```
+
+    ## Simple Example
+
+    ```crystal
+    # src/operations/save_post.cr
+    class SavePost < Post::SaveOperation
+      permit_columns title, body, released_at, draft
+    end
+
+    # src/pages/posts/edit_page.cr
+    class Posts::EditPage < MainLayout
+      needs op : SavePost
+
+      def content
+        form_for(Posts::Update) do
+          para do
+            label_for(@op.title)
+            text_input(@op.title)
+            error_for(@op.title)
+          end
+
+          para do
+            label_for(@op.body)
+            textarea(@op.body)
+            error_for(@op.body)
+          end
+
+          para do
+            label_for(@op.released_at)
+            datetime_input(@op.released_at)
+            error_for(@op.released_at)
+          end
```

Comment:
Could we shorten this to 2 or 3? I know later on it shows how to make it more concise with the Shared::Field component, but I think this could still be a tad bit shorter